Sources/Datadog/Core/SessionTracer.swift

```diff
@@ -0,0 +1,46 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import _Datadog_Private
+
+extension Datadog {
+    internal typealias RequestInterceptor = (URLRequest) -> URLRequest
+    internal typealias TaskObserver = (URLSessionTask) -> Void
+
+    static func trace(_ session: URLSession) throws {
+        let requestInterceptor: RequestInterceptor = { originalRequest in
+            // TODO: RUMM-300 Set tracer HTTP header values
+            return originalRequest
+        }
+        let taskObserver: TaskObserver = { task in
+            // TODO: RUMM-300 observe task to call start/stopSpan
+            /*
+            Pseudo-code:
+            let observation = task.observe(\.state) { observedTask,_ in
+                if detectStartEvent() {
+                    startSpan()
+                } else if detectStopEvent() {
+                    stopSpan()
+                }
+            }
+            objc_setAssociatedObject(
+                task,
+                "__dd_stateObservation",
+                observation,
+                .OBJC_ASSOCIATION_RETAIN_NONATOMIC
+            )
+             */
+        }
+
+        try Swizzler.swizzle(
+            session,
+            requestInterceptor: requestInterceptor,
+            taskObserver: taskObserver,
+            enforceDynamicClassCreation: false
```

Comment:
1. If you swizzle `let obj1: Foo`, its dynamic class will always be `__Datadog_Foo`
2. if you swizzle `let obj2: Foo`, `Swizzler` will lookup `__Datadog_Foo` instead of creating a new class

This is not a problem **as long as you use the same `requestInterceptor`/`taskObserver`s**

`enforceDynamicClassCreation` is an attempt to make this case more explicit: if it is `true`, fetching previously created dynamic classes will throw

Another solution could be creating a new class for each swizzled object, but that wouldn't be practical for our actual use case here.

Reply:
The comment below also addresses the issue here 🙌 
Without `requestInterceptor/taskObserver`, we don't need `enforceDynamicClassCreation` neither 🎉